src/fmu/tools/rms/set_data_constant.py

```diff
@@ -0,0 +1,205 @@
+        dictionary, the method will apply to all properties within these grid
+        models.
+    """
+
+    import roxar
+
+    assert isinstance(config, dict), "Argument must be a Python dictionary!"
+    assert "project" in config.keys(), "Input dict must contain key 'project'!"
+    project = config["project"]
+    assert "value" in config.keys(), "Input dict must contain key 'value'!"
+    value = config["value"]
+
+    def set_safe_value(project, surf_type, name, data_type, value):
+        """
+        Method to set the horizon or zone surface to the defined value.
+        Returns True if the operation succeeded, False otherwise.
+        """
+        success = False
+        try:
+            if surf_type == "horizon":
+                surf = project.horizons[name][data_type]
+            else:
+                surf = project.zones[name][data_type]
+            grid2D = surf.get_grid()
+            grid2D.set_values(grid2D.get_values() * 0.0 + value)
+            surf.set_grid(grid2D)
+            success = True
+            print(" >> >> " + name)
+        except Exception as e:
+            print(" >> >> " + name + " cannot be modified")
+            print(e)
+        return success
+
+    # HORIZON DATA
+    if "horizons" in config.keys():
+        print("Set horizons values to " + str(value) + "...")
+        if isinstance(config["horizons"], list):
+            # work directly at horizon category level
+            for data_type in config["horizons"]:
+                print(" >> " + data_type)
+                for horizon in project.horizons:
+                    set_safe_value(project, "horizon", horizon.name, data_type, value)
+
+        elif isinstance(config["horizons"], dict):
+            # check setup for each horizon category (list vs. all)
+            hor_cat = config["horizons"].keys()
+            for data_type in hor_cat:
+                print(" >> " + data_type)
+                horizons = config["horizons"][data_type]
+                if isinstance(horizons, str):
+                    if horizons == "all":
+                        for horizon in project.horizons:
+                            set_safe_value(
+                                project,
+                                "horizon",
+                                horizon.name,
+                                data_type,
+                                value,
+                            )
+                    else:
+                        assert False, (
+                            "keyword '" + horizons + "' not recognized, 'all' expected!"
+                        )
+                elif isinstance(horizons, list):
+                    for horizon in horizons:
+                        set_safe_value(project, "horizon", horizon, data_type, value)
+
+        else:
+            assert False, (
+                "Value associated with key 'horizons' must be of " "type list or dict!"
+            )
```

Comment:
This section is very similar to the one below for zones, and could be in a separate function.
The only difference is use of `project.horizons, project.zones`, which can be retrieved as `getattr(project, <category>)`, where category is e.g. zones or horizons.